src/packages/auth/src/user-profile.ts

```diff
@@ -0,0 +1,44 @@
+import { BaseDataEntity } from '@exogee/graphweaver';
+import { AuthProvider } from './authentication/base-auth-token-provider';
+
+export type UserProfileData = {
+	provider: AuthProvider;
+	id?: string;
+	displayName?: string;
+	name?: {
```

Comment:
is the name object breakdown necessary/useful to us?

Reply:
I just wanted to give maximum flexibility as you never know what the structure of a user might be.